internal/dag/builder_test.go

```diff
@@ -217,7 +217,7 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 		Spec: gatewayapi_v1alpha1.GatewaySpec{
 			Listeners: []gatewayapi_v1alpha1.Listener{{
 				Port:     80,
-				Protocol: "HTTP",
+				Protocol: "ingress_http",
```

Comment:
Couple more in this section

I guess we're not yet validating Listener.Protocol which would help catch this in unit tests

Reply:
This would normally be caught by the api validation. But since we're skipping that it's missed. Could break this out into helper functions, but until we actually parse this field, it's not validated.